lib/plugins/inspec-suggest/lib/inspec-suggest/suggtool/fetch.rb

```diff
@@ -7,16 +10,26 @@ def fetch(*requested_sets)
       setup(requested_sets)
 
       with_requested_sets do |set_name, set_cfg|
-        # construct expected checkout dir
-        # mkdir -p
-        # for each checkout
-        # check for expected directory
-        # if does not exist
-        #   git clone
-        # else
-        #   git fetch
-        # git checkout onto requested ref
-        #
+        set_dir = File.join(@working_dir, set_name)
+        FileUtils.mkdir_p(set_dir)
+        ui.info "fetch: working on set #{set_name}"
+
+        set_cfg["fetch"].each do |fetch_cfg|
+          checkout_dir = File.join(set_dir, (File.basename(fetch_cfg["repo"], ".git")))
+          git = nil
+          if File.exist?(File.join(checkout_dir, ".git"))
+            git = Git.open(checkout_dir)
+            git.fetch
+            ui.info "fetch: updated"
+          else
+            git = Git.clone(fetch_cfg["repo"], checkout_dir)
+            ui.info "fetch: cloned from #{fetch_cfg["repo"]}"
+          end
+
+          ref = fetch_cfg["ref"] || "main"
+          git.checkout(ref)
+          ui.info "fetch: checked out #{ref}"
+        end
```

Comment:
Do we require any error handling for git connection or error while fetching the repo from git? Or we are ignoring that since this tool will be used internally?

Reply:
I agree, there is no error checking. BUt since it will be used internally and will spew errors as backtraces to STDERR I think we can tolerate it.